src/main/java/net/mcreator/workspace/WorkspaceFileManager.java

```diff
@@ -181,13 +181,7 @@ public void setDataSavedListener(DataSavedListener listener) {
 		void dataSaved();
 	}
 
-	private static class SaveTask implements Runnable {
-
-		private final WorkspaceFileManager fileManager;
-
-		SaveTask(WorkspaceFileManager fileManager) {
-			this.fileManager = fileManager;
-		}
+	private record SaveTask(WorkspaceFileManager fileManager) implements Runnable {

```

Comment:
Record is meant for "data classes", not for classes that do actual logic, this class is not suitable for record from semantic point of view

Reply:
I figured that would shorten the code by few lines. It could suppress compiler warnings as well if you want:
```java
@SuppressWarnings("ClassCanBeRecord")
```

Reply:
No need to suppress warnings in this case I would say, but you can add it